dat/outfits/lib/alone.lua

```diff
@@ -0,0 +1,20 @@
+
+local fmt = require "format"
+
+local prvdesc=descextra
+descextra=function ( p, o, po)
+   return prvdesc( p, o, po) .. "\n#b".."This outfit only works alone.".."#0"
+end
+
+function onoutfitchange( p, po )
+   if p and po then
+      local o = p:outfitSlot('engines_secondary')
+      if o and p:outfitRmSlot('engines_secondary') and p==player.pilot() then
+         player.outfitAdd(o)
+         print(fmt.f("{name} was removed as {myself} refuses its presence.",{
+            name=o:nameRaw(),myself=po:outfit():nameRaw()
+         }))
+      end
+   end
+end
```

Comment:
```suggestion
local slotname = 'engines_secondary'
function onoutfitchange( p, po )
   if p and po then
      local o = p:outfitSlot(slotname)
      if o and p:outfitRmSlot(slotname) and p==player.pilot() then
         player.outfitAdd(o)
         --print(fmt.f("{name} was removed as {myself} refuses its presence.",{
            name=o:nameRaw(),myself=po:outfit():nameRaw()
         }))
      end
   end
end
```

I understand you want to remove the secondary slot always, and only add it if it's the player right?

Reply:
Yep. The player should get its outfit back.